NEWS.md

```diff
@@ -210,6 +223,40 @@
     #[1] TRUE
     ```
 
+18. New `frolladapt` helper function has been added to aid in preparation of adaptive length of rolling window width when dealing with _irregularly spaced ordered data_. This lets the user to apply a rolling function over a period without having to deal with gaps in a data where some periods might be missing, [#3241](https://github.com/Rdatatable/data.table/issues/3241). Thanks to @jangorecki for implementation.
+```r
+idx = as.Date("2022-10-23") + c(0,1,4,5,6,7,9,10,14)
+dt = data.table(index=idx, value=seq_along(idx))
+dt
+#        index value
+#       <Date> <int>
+#1: 2022-10-23     1
+#2: 2022-10-24     2
+#3: 2022-10-27     3
+#4: 2022-10-28     4
+#5: 2022-10-29     5
+#6: 2022-10-30     6
+#7: 2022-11-01     7
+#8: 2022-11-02     8
+#9: 2022-11-06     9
+dt[, c("rollmean3","rollmean3days") := list(
+  frollmean(value, 3),
+  frollmean(value, frolladapt(index, 3), adaptive=TRUE)
```

Comment:
in terms of naming, will this function typically be called inside `froll*` or maybe very nearby (e.g. 1-3 expressions earlier)? it looks slightly redundant to use `froll` twice. `adapt()`/`adaptive()` otoh is maybe too generic, but if we expect >90% of usage to look like your simple example, maybe it's the right choice anyway?

other things that come to mind: `make_adaptive()`/`do_adapt()`, or emphasizing it like `ADAPT()`/`Adapt()`.

Reply:
next, i guess it helps to emphasize somehow that `3` is _in the units of the `index` column_, so that a POSIXct column would have `n` in _seconds_.

Reply:
third, WDYT about `frolladapt()` returning a classed object (or other attribute)?

Then `frollmean()` can just look for that attribute and apply `adaptive=TRUE` automatically.

Now, using `frollmean(., frolladapt(), adaptive=TRUE)` looks pretty redundant because we have to write 'adapt' twice.

Reply:
1. adapt is to generic as it will only be applicable to adaptive rolling functions.
2. I think best to focus on that in documentation rather than in function name
3. very good idea, I will file issue for that, adding it here would require many changes in various places

Reply:
> I think best to focus on that in documentation rather than in function name

Yes, that's what I intended